java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -159,15 +159,8 @@ protected DataAndValidityBuffers allocFixedDataAndValidityBufs(int valueCount, i
     } else {
       // Due to roundup to power-of-2 allocation, the bufferSize could be greater than the
       // requested size. Utilize the allocated buffer fully.;
-      int actualCount = (int) ((bufferSize * 8.0) / (8 * typeWidth + 1));
-      do {
-        validityBufferSize = (int) roundUp8(getValidityBufferSizeFromCount(actualCount));
-        dataBufferSize = (int) roundUp8(actualCount * typeWidth);
-        if (validityBufferSize + dataBufferSize <= bufferSize) {
-          break;
-        }
-        --actualCount;
-      } while (true);
+      validityBufferSize = (int) roundUp8(getValidityBufferSizeFromCount(valueCount));
```

Comment:
@hiteshk25 , can you please give insight into what exactly is the problem here?

Reply:
From the description on jira, there is some concern w.r.t release of extra buffer and I see that it is being released as expected at line#195

Reply:
Before fix bufferSize was 8184. It was demonstrated in the test. After the fix buffer size was just 4128. Test passed with that fix and it also puts 1000 values(valueCount) in IntVector to verify that. Now, it releases all extra allocated memory.

Reply:
I don't think I follow it fully. Are you worried about memory being released properly? If that is the case then memory is being released fine at line#195 AFAIK. Also, I don't see your test case checking for memory leak if that is what your fix is for.

Let's say we allocate an ArrowBuf for 1MB and create two ArrowBufs out of it using slice() of size 256KB each. Now are you suggesting that we are leaking 512KB of memory in the parent buffer?

Reply:
No, it's not a memory leak. It's just to make sure how much buffer we need to store 1000 ints(it just requires 4128 bytes). We don't want to allocate 8184 bytes to store just 1000 ints.

Reply:
That's because of the allocator implementation in Netty that allocates power of 2 chunks. The usual practice has been to use a tuned record count (value count) such that direct memory is allocated optimally for arrow vectors. I believe if you do allocator.getAllocatedMemory() in your test just before the end, it should show 8192

Reply:
Right. With above fix we consume buffer which is required to store 1000 ints. And we release all extra memory.